### ISSUE TITLE
dockerTools broken

### ISSUE THREAD

**Issue by memberbetty** [NixOS/nixpkgs#56209]
## Issue description

When using a nix expression involving `pkgs.dockerTools.buildLayeredImage` where `pkgs` refers to `master`, I get with an expression that worked fine several months ago:

> these derivations will be built:
>   /nix/store/2p2qrri7pd5ill9bbi8yjlz6bzq9aj7l-bulk-layers.drv
>   /nix/store/0kwy8dckl76na45xhfdjmhyhjjqjp7ry-closure.drv
>   /nix/store/8a844y9vymh1x4jrg452c2xc2zxq9gy1-foo-customisation-layer.drv
>   /nix/store/yxbgh36705wsp0bbm78wgxn2wdgga0s3-closure-paths.drv
>   /nix/store/nxx9wh8j8ah9qfcvw5iq71ggh40yyb8d-foo-granular-docker-layers.drv
>   /nix/store/xls3nlawl4jgdc72jn96q244982b2jvv-docker-image-foo.tar.gz.drv
> building '/nix/store/2p2qrri7pd5ill9bbi8yjlz6bzq9aj7l-bulk-layers.drv'...
> building '/nix/store/0kwy8dckl76na45xhfdjmhyhjjqjp7ry-closure.drv'...
> building '/nix/store/8a844y9vymh1x4jrg452c2xc2zxq9gy1-foo-customisation-layer.drv'...
> building '/nix/store/yxbgh36705wsp0bbm78wgxn2wdgga0s3-closure-paths.drv'...
> building '/nix/store/nxx9wh8j8ah9qfcvw5iq71ggh40yyb8d-foo-granular-docker-layers.drv'...
> xargs: /nix/store/jixivxhh3c8sncp9xlkc4ls3y5f2mmxh-store-path-to-layer.sh: Permission denied
> builder for '/nix/store/nxx9wh8j8ah9qfcvw5iq71ggh40yyb8d-foo-granular-docker-layers.drv' failed with exit code 126
> cannot build derivation '/nix/store/xls3nlawl4jgdc72jn96q244982b2jvv-docker-image-foo.tar.gz.drv': 1 dependencies couldn't be built
> error: build of '/nix/store/xls3nlawl4jgdc72jn96q244982b2jvv-docker-image-foo.tar.gz.drv' failed
> 

The `xargs: /nix/store/jixivxhh3c8sncp9xlkc4ls3y5f2mmxh-store-path-to-layer.sh: Permission denied
` stands out. 

> ls -lah /nix/store/jixivxhh3c8sncp9xlkc4ls3y5f2mmxh-store-path-to-layer.sh
> -r--r--r-- 1 root root 608 Jan  1  1970 /nix/store/jixivxhh3c8sncp9xlkc4ls3y5f2mmxh-store-path-to-layer.sh
> 

### Steps to reproduce
Unknown

## Technical details

Treeish: `1d6532d9765b8ba65658e6e6bf4d86b1e4464f8f `

**Comment by memberbetty** [comment]
It appears that this relates to a caching optimization I did. 

> 
> persistent_volume=$(docker volume create myvol) &&
> docker <snip> -v ${persistent_volume}:/nix <more args>

Why wouldn't this work? The general idea is that everything that is ever downloaded ends up in the Nix store on a persistent volume on the host. Disabling this optimization results in a working system.  

For a single version of nixpkgs this optimization worked as planned, but when I updated the treeish to a newer version it instantly broke. That is, the persistent volume does something different than being a plain cache and Nix is sensitive to other files already being present, which suggests some kind of impurity.
